runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java

```diff
@@ -71,14 +72,27 @@ public Token appendCheckpoints(CorfuRuntime rt, String author) {
                                 .getSerializer();
                 cpw.setSerializer(serializer);
 
-                Token minCPSnapshot = cpw.appendCheckpoint();
+                Token minCPSnapshot = Token.UNINITIALIZED;
+                while (retry < numRetries) {
+                    try {
+                        minCPSnapshot = cpw.appendCheckpoint();
+                        break;
+                    } catch (WrongEpochException wee) {
+                        log.info("Epoch changed to {} during append checkpoint snapshot resolution. Sequencer" +
+                                " failover can lead to potential epoch regression, retry {}/{}", wee.getCorrectEpoch(),
+                                retry, numRetries);
+                        retry++;
+                        if (retry == numRetries) {
+                            String msg = String.format("Epochs changed during checkpoint cycle, " +
+                                    "over more than %s times. Potential sequencer regressions can lead to data loss. " +
+                                    "Aborting.", numRetries);
+                            throw new IllegalStateException(msg);
+                        }
+                    }
+                }
 
                 if (minSnapshot == Token.UNINITIALIZED) {
                     minSnapshot = minCPSnapshot;
-                } else if (minSnapshot.getEpoch() != minCPSnapshot.getEpoch()) {
-                    String msg = String.format("Epoch changed during GC cycle from %s to %s", minSnapshot,
-                            minCPSnapshot);
-                    throw new IllegalStateException(msg);
                 } else if (Token.min(minCPSnapshot, minSnapshot) == minCPSnapshot) {
```

Comment:
This min tracking logic can be cleaned up/removed if you decide to do the suggestion above.

Reply:
done.